m4opt/_cli/schedule.py

```diff
@@ -275,7 +282,20 @@ def schedule(
         observer_locations = mission.observer_location(obstimes)
 
     with status("evaluating field of regard"):
-        target_coords = mission.skygrid
+        if isinstance(mission.skygrid, dict):
+            if skygrid is not None:
+                if skygrid not in mission.skygrid:
+                    typer.secho(
+                        f"skygrid '{skygrid}' not found. Options: {', '.join(mission.skygrid.keys())}",
+                        fg=typer.colors.RED,
+                    )
+                    raise typer.Exit(1)
+                target_coords = mission.skygrid[skygrid]
+            else:
```

Comment:
This can be rewritten with fewer nested branches:

```python
if not isinstance(mission.skygrid, dict):
    target_coord = mission.skygrid
elif skygrid in mission.skygrid:
    target_coord = mission.skygrid[skygrid]
else:
    ... # print error, exit
```